From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kevin Raneri <kevin.raneri@gmail.com>
Date: Tue, 30 May 2023 22:22:31 -0400
Subject: [PATCH] Option To Disable End Crystal Dragon Respawn

On servers that expect end crystal fights in the end dimension,
disabling this will prevent the server from performing an expensive
search to attempt respawning the ender dragon whenever a player places
an end crystal.

Note:
I'm tempted to deprecate this as 26.1 made it possible for any dimension
to be able to have the dragon battle (or not have it).
I'll probably keep it due to the ease of access and possibility of toggle
it off with a config reload.

diff --git a/net/minecraft/world/item/EndCrystalItem.java b/net/minecraft/world/item/EndCrystalItem.java
index 1eca60c5a3e1eeb6f44edbb81fe1af011edd8732..89c52329fa4aba982195363b05d206b27e91d294 100644
--- a/net/minecraft/world/item/EndCrystalItem.java
+++ b/net/minecraft/world/item/EndCrystalItem.java
@@ -50,7 +50,7 @@ public class EndCrystalItem extends Item {
                         level.addFreshEntity(crystal);
                         level.gameEvent(context.getPlayer(), GameEvent.ENTITY_PLACE, above);
                         EnderDragonFight fight = ((ServerLevel)level).getDragonFight();
-                        if (fight != null) {
+                        if (fight != null && gg.pufferfish.pufferfish.PufferfishConfig.ALLOW_END_CRYSTAL_RESPAWN) { // Pufferfish
                             fight.tryRespawn(above); // Paper - Perf: Do crystal-portal proximity check before entity lookup
                         }
                     }
